src/features/import/utils/parseFile.ts

```diff
@@ -0,0 +1,60 @@
+import * as XLSX from 'xlsx';
+import { parse } from 'papaparse';
+
+export type ImportedFile = Array<Sheet>;
+type Sheet = { [key: string]: Array<Rows> };
+type Rows = Array<Record<string | number, string | number>>;
+
+export async function parseCSVFile(file: File): Promise<ImportedFile> {
+  return new Promise((resolve, reject) => {
+    const rawData: ImportedFile = [];
+    const reader = new FileReader();
+    reader.readAsDataURL(file);
+
+    reader.onloadend = function () {
+      parse(file, {
+        complete: (result) => {
+          if (result.data) {
+            const sheetObject = {
+              //use name of file as key
+              [file.name.split('.csv').join('')]: result.data as Rows[],
```

Comment:
I'm confused by the `Rows` type, which has a plural name indicating that it's a type that constitutes multiple rows, but here the data is cast to _array of `Rows`_ which sounds to me like a 2D array and I don't see why that would be necessary for a spreadsheet.